src/nspawn/nspawn-mount.c

```diff
@@ -600,6 +610,47 @@ int mount_custom(
         return 0;
 }
 
+// Retrieve existing subsystems. This function is called in a new cgroup namespace.
+static int get_controllers(Set *subsystems)
+{
```

Comment:
`cg_kernel_controllers` returns available controllers and doesn't give us any information about co-mounted controllers. `get_subsystems` simply parses `/proc/self/cgroup` inside the new cgroup namespace. This way we can mount all co-mounted controllers and then symlink the individual controllers that are co-mounted to the co-mounted. We can't get that information reliably just from `get_kernel_subsystems` since we are in a new cgroup namespace and cannot access cgroup information on the host.